File: face-detection/ibm-watson-visual-recognition/IbmWatsonVisualRecognitionFaceDetection.kt
// "input" can be a File or an InputStream object representing the input image.

val iamOptions = IamOptions.Builder()
    .apiKey(apiKey)
    .build()

val client = VisualRecognition("2018-03-16", iamOptions)

val detectFacesOptions = DetectFacesOptions.Builder()
    .imagesFile(input)
    .build()

val result = client
    .detectFaces(detectFacesOptions)
    .execute()

File: face-detection/ibm-watson-visual-recognition/ibm-watson-visual-recognition-face-detection.js
// "input" is a string holding the path to an image file.

const VisualRecognitionV3 = require('watson-developer-cloud/visual-recognition/v3')
const fs = require('fs')

const visualRecognition = new VisualRecognitionV3({
    url: 'https://gateway.watsonplatform.net/visual-recognition/api',
    version: '2018-03-19',
    iam_apikey
})

const params = {
    images_file: fs.createReadStream(input)
}

visualRecognition.detectFaces(params, (err, res) => {
    const faces = res.images[0].faces
})
